src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -1,69 +1,104 @@
-import jsonwebtoken from 'jsonwebtoken';
 import {
   CredentialResolver,
+  isEIP191Jwt,
+  isVerifiableCredential,
   RoleEIP191JWT,
+  transformClaim,
   VerifiableCredential,
 } from '@energyweb/vc-verification';
-import { ClaimData } from '@energyweb/vc-verification/dist/src/models';
+import { RolePayload } from '@energyweb/vc-verification';
 import { RoleCredentialSubject } from '@energyweb/credential-governance';
-import { IPublicClaim } from '@ew-did-registry/claims';
-import { ClaimService } from '../services';
-import { RegistrationTypes } from '../claim.types';
+import * as jwt from 'jsonwebtoken';
+import { DIDService } from '../../did/did.service';
 
 export class RoleCredentialResolver implements CredentialResolver {
-  constructor(private readonly claimService: ClaimService) {}
+  constructor(private readonly didService: DIDService) {}
+
   async getCredential(
     did: string,
     namespace: string
   ): Promise<
     VerifiableCredential<RoleCredentialSubject> | RoleEIP191JWT | null
   > {
+    const tokens = await this.didService.resolveServiceEndpoints(did);
     return (
-      (await this.getVerifiableCredential(did, namespace)) ||
-      (await this.getEIP191JWT(did, namespace))
+      this.serviceEndpointsToCredentials(tokens).find(
+        (cred) => cred.credentialSubject.role.namespace === namespace
+      ) ||
+      this.serviceEndpointsToEIP191(tokens).find(
+        (token) => token.payload.claimData.claimType === namespace
+      )
     );
   }
 
   async getVerifiableCredential(
     subject: string,
-    claimType: string
+    namespace: string
   ): Promise<VerifiableCredential<RoleCredentialSubject> | null> {
-    const claim = await this.claimService.getByClaimType({
-      subject,
-      claimType,
-    });
-
-    return claim?.vp
-      ?.verifiableCredential?.[0] as VerifiableCredential<RoleCredentialSubject>;
+    return (await this.credentialsOf(subject)).find(
+      (token) => token.credentialSubject.role.namespace === namespace
```

Comment:
- Again, probably safer to use optional chaining
- I think credential is a clearer name
```suggestion
      (credential) => credential?.credentialSubject?.role?.namespace === namespace
```